x-pack/plugins/monitoring/server/lib/metrics/apm/metrics.js

```diff
@@ -39,6 +40,32 @@ export const metrics = {
     ),
     field: 'beats_stats.metrics.beat.cpu.total.value',
   }),
+  apm_cgroup_cpu: new QuotaMetric({
+    app: 'apm',
+    ...ApmMetric.getMetricFields(),
+    fieldSource: 'beats_stats.metrics.beat.cgroup',
+    usageField: 'cpuacct.total.ns',
+    periodsField: 'cpu.stats.periods',
+    quotaField: 'cpu.cfs.quota.us',
+    field: 'beats_stats.metrics.beat.cpu.total.value', // backup field if quota is not configured
```

Comment:
What does this backup field actually do? I don't see any data showing up when no quota is configured, which I believe is not a great experience, especially since this is the only CPU related graph for APM.
For ES I see CPU data without having any quota configured - does it fall back to system metrics?

Reply:
That's a good point. I think ES monitoring documents _always_ contain the backup field, but perhaps APM metrics work differently so we'd have to vary our approach here.

Reply:
I checked and the `beats_stats.metrics.beat.cpu.total.value` is actually populated when running inside a docker container. But no CPU values are displayed. 
<img width="756" alt="Screenshot 2021-02-15 at 19 11 21" src="https://user-images.githubusercontent.com/5555349/107981121-ca663800-6fc1-11eb-9513-0abe486eca20.png">
<img width="867" alt="Screenshot 2021-02-15 at 19 12 57" src="https://user-images.githubusercontent.com/5555349/107981146-d4883680-6fc1-11eb-9545-8c49631814b5.png">

Reply:
@simitt This is because there's no mapping for `beats_stats.metrics.beat.cgroup`, and thus we can't do aggregations on these fields.

It needs to be added to this template:
```JSON
GET _template/.monitoring-beats
```
@axw Not sure if this needs to be delegated to Beats or ES team?

Reply:
@igoristic the mapping was added for `7.11` https://github.com/elastic/elasticsearch/pull/65997

Reply:
Sorry, missed that ticket

I'm missing the whole `cgroup` object:
https://gist.github.com/igoristic/dcf8e5257628c7212553a53b17ead60c

Reply:
@igoristic maybe pull a new image? Or use 7.11.0 GA? I have a 7.11.0 deployment in Cloud, and it definitely has the cgroup object: https://gist.github.com/axw/c0e0db05ba8756b26a45f40c4f7f697c

Reply:
Yea I see it on cloud too. I'd double check your local ES is a fresh copy

Reply:
I don't think the backup logic is actually working so we might need a change like this:

```
--- a/x-pack/plugins/monitoring/server/lib/metrics/classes/quota_metric.js
+++ b/x-pack/plugins/monitoring/server/lib/metrics/classes/quota_metric.js
@@ -55,6 +55,7 @@ export class QuotaMetric extends Metric {
       const quota = get(bucket, 'quota.value');
       const deltaUsageDerivNormalizedValue = get(bucket, 'usage_deriv.normalized_value');
       const periodsDerivNormalizedValue = get(bucket, 'periods_deriv.normalized_value');
+      const backupMetricDerivNormalizedValue = get(bucket, 'metric_deriv.normalized_value');
 
       if (deltaUsageDerivNormalizedValue && periodsDerivNormalizedValue && quota > 0) {
         // if throttling is configured
@@ -63,7 +64,7 @@ export class QuotaMetric extends Metric {
         return factor * 100; // convert factor to percentage
       }
       // if throttling is NOT configured, show nothing. The user should see that something is not configured correctly
-      return null;
+      return backupMetricDerivNormalizedValue || null;
     };
   }
 }
```